lib/internal/source_map/source_map.js

```diff
@@ -241,7 +242,13 @@ class SourceMap {
           break;
       }
 
-      columnNumber += decodeVLQ(stringCharIterator);
+      // A negative columnOffset is valid, and if one is observed sorting is
+      // necessary, see: https://github.com/mozilla/source-map/pull/92
+      const columnOffset = decodeVLQ(stringCharIterator);
+      if (columnOffset < 0) {
+        hadNegativeColumnOffset = true;
+      }
+      columnNumber += columnOffset;
       if (isSeparator(stringCharIterator.peek())) {
```

Comment:
How much can we clean up this code? The whole iterator string iterator wrapper thing is icky to me.

Reply:
we've diverged quite a bit form the V8 implementation at this point, introducing private fields as an example. I probably wouldn't do it in this PR, but I think you could go ahead and refactor as much as you see fit.